pkg/auth/cache.go

```diff
@@ -7,9 +7,12 @@ import (
 	"github.com/treeverse/lakefs/pkg/cache"
 )
 
-type CredentialSetFn func() (*model.Credential, error)
-type UserSetFn func() (*model.User, error)
-type UserPoliciesSetFn func() ([]*model.Policy, error)
+type (
+	CredentialSetFn     func() (*model.Credential, error)
+	UserSetFn           func() (*model.User, error)
+	UserPoliciesSetFn   func() ([]*model.Policy, error)
+	ExternalPrincipalFn func() (*model.ExternalPrincipal, error)
+)
```

Comment:
from a quick look it seems like we need to start using generics here

Reply:
Elaborate? 
I explored several options here but the small changes don't seem very valuable and others i had required larger design that i prefer to be out of scope.

Reply:
not a blocker. but it looks like a copy paste for each type we like to cache

Reply:
It's indeed the case. 
I'm comfortable with this since it happens once in a blue moon. 
But open to suggestions